backend/database.js

```diff
@@ -172,6 +173,51 @@ export default (sbp('sbp/selectors/register', {
   }
 }): any)
 
+if (persistence === 'fs') {
+  sbp('sbp/selectors/register', {
+    'backend/db/readString': async function (key: string): Promise<string | null> {
+      const bufferOrError = await sbp('backend/db/readFile', key)
+      if (Boom.isBoom(bufferOrError)) {
+        return null
+      }
+      return bufferOrError.toString('utf8')
+    },
+    'backend/db/writeString': async function (key: string, value: string): Promise<void> {
+      // eslint-disable-next-line no-useless-catch
+      try {
+        return await sbp('backend/db/writeFile', key, value)
+      } catch (err) {
+        throw err
+      }
+    }
+  })
+} else if (persistence === 'sqlite') {
+  sbp('sbp/selectors/register', {
+    'backend/db/readString': function (key: string): Promise<string> {
+      return new Promise((resolve, reject) => {
+        db.get('SELECT value FROM Strings WHERE key = ?', [key], (err, row) => {
+          if (err) {
+            reject(err)
+          } else {
+            resolve(row?.value)
+          }
```

Comment:
What happens if the key is not found? Is an error thrown or does it resolve as `null`?

I ask because the previous behavior needs to be preserved, as selectors like `'backend/db/lookupName'` depend on it.

Reply:
OK, answered my own question, it returns `undefined`. This will be important later on, see my other comment.